django_auth_ldap/backend.py

```diff
@@ -190,6 +193,41 @@ def get_group_permissions(self, user, obj=None):
 
         return permissions
 
+    def get_user_permissions(self, user, obj=None):
+        if not hasattr(user, "ldap_user") and self.settings.AUTHORIZE_ALL_USERS:
+            _LDAPUser(self, user=user)  # This sets user.ldap_user
+
+        if hasattr(user, "ldap_user"):
+            permissions = self._get_permissions(user, obj, "user")
+        else:
+            permissions = set()
+
+        return permissions
+
+    def _get_permissions(self, user_obj, obj, from_name):
+        """
+        Return the permissions of `user_obj` from `from_name`. `from_name` can
+        be either "group" or "user" to return permissions from
+        `_get_group_permissions` or `_get_user_permissions` respectively.
+        """
+        if not user_obj.is_active or user_obj.is_anonymous or obj is not None:
```

Comment:
When obj is not None, the backend should return the user permission for that object specifically, not an empty set.